src/position.cpp

```diff
@@ -311,6 +292,27 @@ void Position::set_castling_right(Color c, Square rfrom) {
 }
 
 
+/// Position::set_check_info() sets king attacks to detect if a move gives check
+
+void Position::set_check_info(CheckInfo* ci) const {
+
+  ci->blockers[WHITE] = slider_blockers(pieces(BLACK), square<KING>(WHITE));
+  ci->blockers[BLACK] = slider_blockers(pieces(WHITE), square<KING>(BLACK));
+  
+  ci->pinned       = pieces(sideToMove) & ci->blockers[sideToMove];
+  ci->dcCandidates = pieces(sideToMove) & ci->blockers[~sideToMove];
```

Comment:
ci->pinned and ci->dcCandidates seems redundant with ci->blockers[]. Why keep them all? Can you drop pinned and dcCandidates?